src/cli.js

```diff
@@ -0,0 +1,276 @@
+        util.handleError("stdin", e);
+      }
+    });
+  }
+
+  function formatFiles() {
+    eachFilename(filepatterns, (filename, options) => {
+      if (argv["write"]) {
+        // Don't use `console.log` here since we need to replace this line.
+        process.stdout.write(filename);
+      }
+
+      let input;
+      try {
+        input = fs.readFileSync(filename, "utf8");
+      } catch (e) {
+        // Add newline to split errors from filename line.
+        process.stdout.write("\n");
+
+        console.error(`Unable to read file: ${filename}\n${e}`);
+        // Don't exit the process if one file failed
+        process.exitCode = 2;
+        return;
+      }
+
+      listDifferent(input, options, filename);
+
+      const start = Date.now();
+
+      let result;
+      let output;
+
+      try {
+        result = format(
+          input,
+          Object.assign({}, options, { filepath: filename })
+        );
+        output = result.formatted;
+      } catch (e) {
+        // Add newline to split errors from filename line.
+        process.stdout.write("\n");
+
+        util.handleError(filename, e);
+        return;
+      }
+
+      if (argv["write"]) {
+        // Remove previously printed filename to log it with duration.
+        readline.clearLine(process.stdout, 0);
+        readline.cursorTo(process.stdout, 0, null);
+
+        // Don't write the file if it won't change in order not to invalidate
+        // mtime based caches.
+        if (output === input) {
+          if (!argv["list-different"]) {
+            console.log(chalk.grey("%s %dms"), filename, Date.now() - start);
+          }
+        } else {
+          if (argv["list-different"]) {
+            console.log(filename);
+          } else {
+            console.log("%s %dms", filename, Date.now() - start);
+          }
+
+          try {
+            fs.writeFileSync(filename, output, "utf8");
+          } catch (err) {
+            console.error(`Unable to write file: ${filename}\n${err}`);
+            // Don't exit the process if one file failed
+            process.exitCode = 2;
+          }
+        }
+      } else if (argv["debug-check"]) {
+        if (output) {
+          console.log(output);
+        } else {
+          process.exitCode = 2;
+        }
+      } else if (!argv["list-different"]) {
+        util.writeOutput(result, options);
+      }
+    });
+  }
+
+  function getOptionsForFile(filePath) {
+    const options =
+      argv["config"] === false ? null : resolver.resolveConfig.sync(filePath);
+
+    try {
+      const parsedArgs = minimist(args, {
+        boolean: constant.booleanOptionNames,
+        string: constant.stringOptionNames,
+        default: Object.assign(
+          {
+            semi: true,
+            "bracket-spacing": true,
+            parser: "babylon"
+          },
+          util.dashifyObject(options)
+        )
+      });
+      return util.getOptions(Object.assign({}, argv, parsedArgs));
```

Comment:
Existing issue, but if this throws, the error message `"Invalid configuration file"` will be incorrect. Maybe just change the string to `"Invalid configuration:"`?